src/Bitmap_cubical_complex/include/gudhi/Bitmap_cubical_complex_base.h

```diff
@@ -517,46 +638,46 @@ class Bitmap_cubical_complex_base {
   void read_perseus_style_file(const char* perseus_style_file);
   void setup_bitmap_based_on_top_dimensional_cells_list(const std::vector<unsigned>& sizes_in_following_directions,
                                                         const std::vector<T>& top_dimensional_cells);
+  void setup_bitmap_based_on_vertices(const std::vector<unsigned>& sizes_in_following_directions,
+                                      const std::vector<T>& vertices);
   Bitmap_cubical_complex_base(const char* perseus_style_file, std::vector<bool> directions);
   Bitmap_cubical_complex_base(const std::vector<unsigned>& sizes, std::vector<bool> directions);
-  Bitmap_cubical_complex_base(const std::vector<unsigned>& dimensions, const std::vector<T>& top_dimensional_cells,
-                              std::vector<bool> directions);
+  Bitmap_cubical_complex_base(const std::vector<unsigned>& dimensions, const std::vector<T>& cells,
+                              std::vector<bool> directions, const bool& input_top_cells);
 };
 
 template <typename T>
 void Bitmap_cubical_complex_base<T>::put_data_to_bins(std::size_t number_of_bins) {
-  bool dbg = false;
 
   std::pair<T, T> min_max = this->min_max_filtration();
   T dx = (min_max.second - min_max.first) / (T)number_of_bins;
 
   // now put the data into the appropriate bins:
   for (std::size_t i = 0; i != this->data.size(); ++i) {
-    if (dbg) {
-      std::clog << "Before binning : " << this->data[i] << std::endl;
-    }
+#ifdef DEBUG_TRACES
+    std::clog << "Before binning : " << this->data[i] << std::endl;
+#endif
```

Comment:
I seem to remember that Paweł objected to this, but I don't remember why. Maybe he thought the variable was easier to turn on for one class without turning it on for the whole of gudhi?
Anyway, he hasn't contributed in a while, and I don't mind the change.

Reply:
If I remember well, your explanation is the good one. IMHO when you want debug traces, it is ok if you have the whole gudhi debug traces, specifically because of our modular way of programming.